File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/data/framework/multipart/IMultipartConverter.java
package ru.usedesk.chat_sdk.internal.data.framework.multipart;

import okhttp3.MultipartBody;
import ru.usedesk.chat_sdk.internal.data.framework.file.entity.LoadedFile;

public interface IMultipartConverter {
    MultipartBody.Part convert(String key, String value);

    MultipartBody.Part convert(String key, LoadedFile loadedFile);
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/data/repository/api/loader/multipart/IMultipartConverter.kt
package ru.usedesk.chat_sdk.data.repository.api.loader.multipart

import okhttp3.MultipartBody

internal interface IMultipartConverter {
    fun convert(pair: Map.Entry<String, Any?>): MultipartBody.Part?

    fun convert(
        key: String,
        byteArray: ByteArray,
        originalFile: String
    ): MultipartBody.Part
}